Bump Airthings BLE to 1.1.1

LaStrada opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Updating the Airthings BLE library to support Wave Enhance and Corentium Home 2, our two newest devices.

https://github.com/Airthings/airthings-ble/compare/0.9.2...1.1.1

The 1.0.0 version was skipped due to some parsing bugs in some cases.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

LaStrada committed:
Update airthings ble integration to work with latest changes

LaStrada committed:
Update strings

LaStrada committed:
Update version

LaStrada committed:
Config flow bugfix

LaStrada committed:
Update logs

LaStrada committed:
Refactor service uuids

LaStrada committed:
Fix test_bluetooth_discovery test

LaStrada committed:
Revert

LaStrada committed:
Fix tests

LaStrada committed:
Fix title in config flow

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

LaStrada commented:
The lib needs an update due to some errors. Need to fix a few things:
https://github.com/home-assistant/core/actions/runs/17668415272/job/50214862409?pr=147539

Will take another look at it

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

LaStrada committed:
Bump library

LaStrada committed:
Add more tests

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

bdraco commented:
For config flows, we need to have 100% coverage before merging. Thanks!

LaStrada committed:
More tests

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

bdraco committed:
Merge branch 'dev' into update-airthings-ble-framework

bdraco commented:
Unless I missed it, it doesn't look like there are breaking changes in the library, can you split the library bump into a separate PR. If there are breaking changes they can be in that same PR.  Finally if its can't be all of this needs to be merged together, please add an explanation as to why in the opening text.

LaStrada commented:
> Unless I missed it, it doesn't look like there are breaking changes in the library.. can you split the library bump into a separate PR? If there are breaking changes they can be in that same PR. Finally if its can't be all of this needs to be merged together, please add an explanation as to why in the opening text.

There aren't really many changes in the library for the previously supported devices. The main two things are:

1. Better handling of unsupported devices. Technically this has always been an issue, but the newer generations can look the same if we don't get the device name. It's the same BLE service and so on. So in some cases HA thinks the View Plus/Radon/Pollution is supported, but it's not (WiFi sync only).
2. Added support for Wave Enhance and Corentium Home 2 (which is related to previous bullet point) and the main reason for the library bump.
3. Some devices (for now, only the new devices) needs an FW upgrade before we can add it to HA. Added some code for this so we can ask the user to upgrade the FW before adding it. The released FW version of Wave Enhance doesn't really give us enough information over BLE to be really useful.

There is already 100% test coverage for config flow, so the new tests are added to be able to get 100% again.

The only things I see that can be moved out is:
* The new sensor types in `sensor.py` (will move it out and open a PR with it after this one is merged)
* A bugfix in `config_flow.py` (PR opened)
* Name formatting in config flow (PR merged)
* Maybe some of the logging (ongoing)

And after all this, I can fix the wrong type you pointed out in the library.

bdraco commented:
Is there any reason the library cannot be bumped in a separate PR and than this one merged after?

LaStrada commented:
My initial thought was that it will start finding unsupported devices. But if I skip the changes in the manifest file / service UUIDs in config flow, it might work 🤔 Will try again tomorrow

LaStrada committed:
Merge branch 'dev' into update-airthings-ble-framework

LaStrada commented:
I'm closing this and starts over again. Will open a PR to only bump the library, then one with Wave Enhance / Corentium Home 2 support.
